### PR TITLE
chore(dependency): update carbon-components dependency

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

Expressive theme – Multi select #1010
Expressive theme – time picker #1085
Expressive theme – Progress indicator #1039

### Description

Bump up `carbon-components` dependency in styles package to get the latest bug fixes.

### Changelog

**Changed**

- bump `carbon-components` dependency

### CODE REVIEW

File: packages/styles/package.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
     "@carbon/layout": "10.7.0",
     "@carbon/themes": "10.9.0",
     "@carbon/type": "10.5.1",
-    "carbon-components": "10.9.0"
+    "carbon-components": "10.9.3-rc.0"
```
Comment (jeffchew) [inline packages/styles/package.json]: @annawen1 is there a date when they are doing the full `10.9.3` release? We probably should create a followup story to upgrade once it's out of rc.

Comment (annawen1) [reply packages/styles/package.json]: hmm based on the comments in the [bug ticket](https://github.com/carbon-design-system/carbon/issues/5073) doesn't seem like they have a date in mind yet

Comment (jeffchew) [reply packages/styles/package.json]: Should we potentially hold off on upgrading this then? This update affects all components/patterns.

Comment (annawen1) [reply packages/styles/package.json]: ok I'm good with waiting for full release

Comment (annawen1) [reply packages/styles/package.json]: reopening this PR with the latest carbon dependency versions

**Comment by annawen1** [comment]
will need to check all patterns and components to make sure things aren't broken

**Comment by jeffchew** [comment]
> will need to check all patterns and components to make sure things aren't broken

Did a quick glance through of the production components/patterns and didn't see anything glaring. The QA team can probably do a more thorough look through in the next regression cycle (for v1.6.0).

**Pull request merged**